selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestProjectServiceClient.java

```diff
@@ -221,53 +215,51 @@ public void updateFile(String workspaceId, String pathToFile, String content) th
 
   public boolean checkProjectType(String wokspaceId, String projectName, String projectType)
       throws Exception {
-    return getProjectParameters(wokspaceId, projectName)
-        .get("type")
-        .getAsString()
-        .equals(projectType);
+    return getProjectConfig(wokspaceId, projectName).getType().equals(projectType);
   }
 
-  public boolean checkProjectLanguage(String wokspaceId, String projectName, String projectLanguage)
+  public boolean checkProjectLanguage(String workspaceId, String projectName, String language)
       throws Exception {
-    return getProjectParameters(wokspaceId, projectName)
-        .getAsJsonObject("attributes")
+
+    return getProjectConfig(workspaceId, projectName)
+        .getAttributes()
+        .get("language")
+        .contains(language);
+  }
+
+  public boolean checkProjectLanguage(
+      String workspaceId, String projectName, List<String> languages) throws Exception {
+
+    return getProjectConfig(workspaceId, projectName)
+        .getAttributes()
         .get("language")
-        .getAsString()
-        .equals(projectLanguage);
+        .containsAll(languages);
   }
 
   public List<String> getExternalLibraries(String workspaceId, String projectName)
       throws Exception {
-    List<String> result = new ArrayList<>();
     HttpJsonResponse response =
         requestFactory
             .fromUrl(workspaceAgentApiEndpoint.get(workspaceId) + "java/navigation/libraries")
             .useGetMethod()
             .addQueryParam("projectpath", "/" + projectName)
             .request();
 
-    parseResponseAsArray(response)
-        .forEach(
-            jsonElement -> result.add(jsonElement.getAsJsonObject().get("name").getAsString()));
-
-    return result;
-  }
-
-  private JsonObject parseResponseAsJsonObject(HttpJsonResponse response) {
-    return new JsonParser().parse(response.asString()).getAsJsonObject();
+    return DtoFactory.getInstance()
+        .createListDtoFromJson(response.asString(), ProjectConfigDto.class)
+        .stream()
+        .map(e -> e.getName())
+        .collect(Collectors.toList());
   }
 
-  private JsonArray parseResponseAsArray(HttpJsonResponse response) {
-    return new JsonParser().parse(response.asString()).getAsJsonArray();
-  }
-
-  private JsonObject getProjectParameters(String workspaceId, String projectName) throws Exception {
+  private ProjectConfigDto getProjectConfig(String workspaceId, String projectName)
+      throws Exception {
     HttpJsonResponse response =
         requestFactory
             .fromUrl(workspaceAgentApiEndpoint.get(workspaceId) + "project/" + projectName)
             .useGetMethod()
             .request();
 
-    return parseResponseAsJsonObject(response);
+    return DtoFactory.getInstance().createDtoFromJson(response.asString(), ProjectConfigDto.class);
```

Comment:
**ProjectConfig** is more suitable here as more general type compare to **ProjectConfigDto** class.